### DISCUSSION TITLE
Webpack dev server includes runtime modules into entry bundle

### DISCUSSION THREAD

**Discussion by cut2run** [webpack/webpack#18337] [Q&A] [answered]
I have very simple webpack configuration for one Typescript file as the entry (partial config) ...
``` JavaScript
config.entry = {
    launchEventsJS: {
      import: "./src/taskpane/launchEventsJS.ts",
      filename: getScriptName(dev),
    },
  };
```

And webpack-dev-server configuration as follow ...

``` JavaSctipt
const getDevServerConfig = async (): Promise<webpackDevServer.Configuration> => {
  return {
    headers: {
      "Access-Control-Allow-Origin": "*",
    },
    server: {
      type: "https",
      options: await devCerts.getHttpsServerOptions(),
    },
    port: 3000,
  };
}
```

When I run this configuration with webpack command `webpack --mode development` I am getting the expected result of a small transpiled JavaScript file (This is exactly what I need) ...
[![enter image description here][1]][1]

At the same time, when I run the same configuration with webpack-dev-server to run dev server and host the JavaScript file with the command `"webpack-dev-server --mode development` I am getting the bundle file which includes dev server dependencies into transpiled JavaScript file ...
[![enter image description here][2]][2]

How can I get rid of the runtime dependencies to be included with the transpiled JavaScript file? Are there any `webpack-dev-server` configuration settings that must be added?

  [1]: https://i.stack.imgur.com/GBLdL.png
  [2]: https://i.stack.imgur.com/jttm7.png

**Comment by alexander-akait** [comment, marked as answer]
Without extra code hot and live reloading will not work, anyway you can disable it using `client: false` and `hot: false`

**Reply by cut2run** [reply]
The suggested settings worked out for my project. I still have the transpile of TypeScript by Webpack and the devserver doesn't add all those libraries. Ironically I have tried `hot: false` myself before I looked for help and it didn't work as the client dependencies were still added. With both of those properties set to false, I've got the result I needed. Thanks.
